test/sqllogictest/transform/filter_index.slt

```diff
@@ -43,6 +43,41 @@ select b, c from foo where a = 5
 this
 -4.400
 
+query T multiline
+explain plan for select b, c from foo, (select 5 as a) bar where foo.a = bar.a
```

Comment:
These all look good, but I'm not sure what they are testing. E.g. if a future PR causes a change here, is that bad? Idk.

Reply:
I'll add comments with what they're testing for, namely that they should get planned as filters and not one-input joins.

If you run these tests against main, the plan will be
```
%0 =
| Get materialize.public.foo (u1)

#%1
| Join %0 (= #0 5)
....
```